public/app/features/dashboard-scene/settings/variables/VariableEditorForm.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { FormEvent } from 'react';
+
+import { SelectableValue, VariableType } from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import { SceneVariable, AdHocFiltersVariable, AdHocFilterSet } from '@grafana/scenes';
+import { VariableHide, defaultVariableModel } from '@grafana/schema';
+import { HorizontalGroup, Button } from '@grafana/ui';
+import { ConfirmDeleteModal } from 'app/features/variables/editor/ConfirmDeleteModal';
+// import { VariableHideSelect } from 'app/features/variables/editor/VariableHideSelect';
+import { VariableHideSelect } from 'app/features/variables/editor/VariableHideSelect';
+import { VariableLegend } from 'app/features/variables/editor/VariableLegend';
+import { VariableTextAreaField } from 'app/features/variables/editor/VariableTextAreaField';
+import { VariableTextField } from 'app/features/variables/editor/VariableTextField';
+import { VariableNameConstraints } from 'app/features/variables/editor/types';
+
+import { VariableTypeSelect } from './components/VariableTypeSelect';
+import { getVariableEditor, getVariableScene, isEditableVariableType } from './utils';
+
+interface VariableEditorFormProps {
+  variable: SceneVariable;
+  onSubmit: (variable: SceneVariable) => void;
+}
+
+export function VariableEditorForm({ variable, onSubmit }: VariableEditorFormProps) {
+  const [editingVariable, setEditingVariable] = React.useState(variable.clone());
```

Comment:
don't think you need this clone

Reply:
This clone is in place to allow a temporal state until you click on `Apply`. When updating the form, the cloned variable is the one updated. When applying the original variable is replaced by the new one.

Because of the destructive action of changing the variable type and losing the previous configuration, we left the apply button as it exists nowadays.

So, when the user updates the form, it saves the changes on that variable when submitting the form.

Reply:
Yeah, think we can skip this clone and manipulate the original object instead. Capturing the state before the edit interaction started should be sufficient to be able to discard changes if needed.

Reply:
Fixed 😄